logs_ingest/filtering.py

```diff
@@ -0,0 +1,117 @@
+#   Copyright 2021 Dynatrace LLC
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import fnmatch
+import json
+import os
+import re
+from typing import Dict, Set
+
+from logs_ingest import logging
+from logs_ingest.mapping import severity_to_log_level_dict, log_level_to_severity_dict, RESOURCE_TYPE_ATTRIBUTE, \
+    RESOURCE_ID_ATTRIBUTE
+
+GLOBAL = "GLOBAL"
+AMOUNT_OF_DOT_SEPARATORS_IN_GLOBAL_FILTERS = 2
+
+
+class LogFilter:
+    def __init__(self):
+        self._filter_config: str = os.environ.get("FILTER_CONFIG", "")
+        filter_config_pattern = re.compile(r'([^;\s].+?)=([^;]*)')
+        self._filters_tuples = filter_config_pattern.findall(self._filter_config)
+        self._filters_tuples = [self._prepare_filters_tuples(filter_tuple) for filter_tuple in self._filters_tuples]
+        self.filters_dict = self._prepare_filters_dict()
+        if self.filters_dict:
+            logging.info(f"Applied filter_config: {self._filter_config}")
+
+    @staticmethod
+    def _prepare_filters_tuples(filter_tuple):
+        filter_name = filter_tuple[0].strip()
+        value = filter_tuple[1].strip()
+        if filter_name.count(".") > AMOUNT_OF_DOT_SEPARATORS_IN_GLOBAL_FILTERS:
```

Comment:
Why not simple prefix check for supported types of group? right now second part of key doesn't matter, `FILTER.TEST.MIN_LOG_LEVEL.X=4` is evaluated the same as `FILTER.RESOURCE_TYPE.MIN_LOG_LEVEL.X=4`, I'm not sure if we want such a undefined behavior. Additionally, if we would introduce additional valid group type (other than `RESOURCE_TYPE` and `RESOURCE_ID`), it couldn't contain a comma.